apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs

```diff
@@ -0,0 +1,188 @@
+
+render :: Self -> JSX
+render self@{ setState, state: { updateState }, props: { creditCards, onCancel } } =
+  DOM.div
+    { className: "clearfix credit-card-update--container"
+    , children:
+        [ case updateState of
+            ChooseCreditCard       -> Choice.choice
+                                        { creditCards: creditCards
+                                        , onSubmit: \creditCard -> Aff.launchAff_ $ registerCreditCard setState self.props self.state creditCard
+                                        , onCancel: onCancel
+                                        }
+
+            RegisterCreditCard url -> Register.register
+                                        { terminalUrl: url
+                                        }
+
+            Cancel                 -> Router.redirect
+                                        { to: { pathname: "/"
+                                              , state: {}
+                                              }
+                                        , from: "/kreditkortt/uppdatera"
+                                        , push: true
+                                        }
+        ]
+    }
+  where
+    title :: JSX
+    title = DOM.h3_ [ DOM.text "Uppdatera ditt kredit- eller bankkort" ]
+
+registerCreditCard :: SetState -> Props -> State -> CreditCard -> Aff Unit
+registerCreditCard setState props@{ logger, onError, setWrapperState } state oldCreditCard = do
+  creditCardRegister <- User.registerCreditCard
+  case creditCardRegister of
+    Right register@{ terminalUrl: Just url } -> do
+      let newState = state { updateState = RegisterCreditCard url }
+      liftEffect $ do
+        setState \_ -> newState
+        setWrapperState _ { closeable = false }
+      void $ Aff.forkAff $ startRegisterPoller setState props newState oldCreditCard register
+    Right register@{ terminalUrl: Nothing } ->
+      liftEffect $ do
+        logger.log "No terminal url received" Sentry.Error
+        onError
+    Left err ->
+      liftEffect $ do
+        logger.log ("Got the following error when registering credit card: " <> bottegaErrorMessage err) Sentry.Error
+        onError
+
+killRegisterPoller :: State -> Aff Unit
+killRegisterPoller state = Aff.killFiber (error "Canceled poller") state.poller
+
+startRegisterPoller :: SetState -> Props -> State -> CreditCard -> CreditCardRegister -> Aff Unit
+startRegisterPoller setState props state oldCreditCard creditCardRegister = do
+  newPoller <- Aff.forkAff do
+    killRegisterPoller state
+    newPoller <- Aff.forkAff $ pollRegister setState props state oldCreditCard (Right creditCardRegister)
+    Aff.joinFiber newPoller
+  liftEffect $ setState _ { poller = newPoller }
+
+pollRegister :: SetState -> Props -> State -> CreditCard -> Either BottegaError CreditCardRegister -> Aff Unit
+pollRegister setState props@{ logger, onError, onSuccess, onCancel } state oldCreditCard (Right register) = do
+  Aff.delay $ Aff.Milliseconds 1000.0
```

Comment:
This doesn't feel like the right place for waiting. We know what the value of `register.status.state` is even before calling `delay`, why not act on it right away? If the value is `CreditCardRegisterCompleted` then won't this just add one more second to the whole process?

Reply:
It's a trick to poll every second, but we can think of a better place for that delay sure.

I never really questioned it because I started this component by copying **vetrina**'s code, but it is certainly good to scrutinize it, cause that delay is indeed annoying when the register operation has succeded.